ui/app/ducks/swaps/swaps.js

```diff
@@ -111,6 +148,42 @@ export const getFetchingQuotes = (state) => state.swaps.fetchingQuotes
 
 export const getQuotesFetchStartTime = (state) => state.swaps.quotesFetchStartTime
 
+export const getSwapsCustomizationModalPrice = (state) => state.swaps.customGas.price
+
+export const getSwapsCustomizationModalLimit = (state) => state.swaps.customGas.limit
+
+export const getSwapGasEstimateLoadingStatus = (state) => state.swaps.customGas.loading
+
+export const getSwapGasPriceEstimateData = (state) => state.swaps.customGas.priceEstimates
+
+export const getSwapsPriceEstimatesLastRetrieved = (state) => state.swaps.customGas.priceEstimatesLastRetrieved
+
+export function isCustomSwapsGasPriceSafe (state) {
+  const { average } = getSwapGasPriceEstimateData(state)
+
+  const customGasPrice = getSwapsCustomizationModalPrice(state)
+
+  if (!customGasPrice) {
+    return true
+  }
+
+  if (average === null) {
+    return false
```

Comment:
We might want to reconsider this, depending on how the failure state is handled. I assume you had the initial/loading state in mind here, so it would be marked as "unsafe" if it rendered while it was loading.

This left me wondering what `isCustomSwapsGasPriceSafe` was meant to communicate. Clearly it should say yes or no if it's _known_ whether the price is safe, but what should it do if we don't know either way? The way this is handled now is mixed - we return `true` or `false` in the unknown state depending on _why_ we don't know.

Reply:
This was also part of eed17631f and 38a1a21e1. A couple comments on the changes made there:

- comparing to `null` was a mistake, it should have been checking if `average === undefined` which is the initial state
- I decided to rename `isCustomSwapsGasPriceSafe` to `isCustomSwapsGasPriceUnSafe` and return false whenever we don't know. Any claim about the safety of a price would be meaningless if we have no grounds to claim so. If `isCustomSwapsGasPriceUnSafe` returns `false`, it means that we either know it is safe or don't know whether safe or unsafe. In both cases, the UX will be to just not show the user any message or indication about the safety or unsafety of a price.